### PR TITLE
Split shadycss

### LINKED ISSUE #46: Factor library so that less code is needed when native shadow dom is available
Currently this library provides (1) a style scoping mechanism mean to provide styling like native shadow dom, (2) a css custom properties shim, (3) a css @apply shim that works with native custom properties, (4) a custom-style element that supports these features outside of shadow dom.

Browsers like Chrome and Safari 10.2 that have native shadow dom and css custom properties still need feature (3) @apply shimming. Ideally, this library would be factored so that these browsers can load a build that provides only this functionality.

### PR DESCRIPTION
**Pull request by azakus**
Split ShadyCSS into multiple entrypoints that provide features

- Scoping Shim: Provides ShadyDOM style scoping integration and an implementation of CSS Custom Properties

- Apply Shim: Provides support for CSS Mixins with `@apply` syntax

- Custom Style Interface: Library that integrates with the above to support document-level stylesheets with `@apply` and/or style scoping.

Fixes #46

### CODE REVIEW

File: src/custom-style-interface.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  getStyleForCustomStyle(customStyle) {
+    let style;
+    if (customStyle['getStyle']) {
+      style = customStyle['getStyle']();
+    } else {
+      style = customStyle;
+    }
+    return style;
+  }
+  findStyles() {
```
Comment (sorvell) [inline src/custom-style-interface.js]: Rename to `processStyles`?

File: src/style-util.js [hunk 2]
```diff
@@ -163,23 +216,25 @@ export function processVariableAndFallback(str, callback) {
...
-  BRACKETED: /\{[^}]*\}/g,
-  HOST_PREFIX: '(?:^|[^.#[:])',
-  HOST_SUFFIX: '($|[.:[\\s>+~])'
-}
+/**
+ * @param {Element|Object} element
+ * @return {{is: string, extends: string}}
+ */
+export function getIsExtends(element) {
+  let is = element.is || (element.getAttribute && element.getAttribute('is')) || element.localName;
```
Comment (sorvell) [inline src/style-util.js]: Check `localName` first?

Comment (azakus) [inline src/style-util.js]: That will not work for type extension elements.

Comment (azakus) [inline src/style-util.js]: I'll just remove the `element.is` an `element.extends` checks, but leave the rest of the line. WDYT?

Comment (azakus) [inline src/style-util.js]: I'll just check that localname has `-` in it. Should be faster than `getAttribute` calls in the hot path

Comment (azakus) [inline src/style-util.js]: Done

Comment (sorvell) [inline src/style-util.js]: seems ok.

**Pull request merged**